docs/src/components/HeadingWithLink.tsx

```diff
@@ -27,17 +46,14 @@ interface Props extends SpaceAfter {
 const HeadingWithLink = ({ children, noId, spaceAfter = "none" }: Props) => {
   const headingText = getTextFromChildren(children);
   const slugifiedText = slugify(headingText);
+
   return (
     <StyledAnchorWrapper id={noId ? "" : slugifiedText}>
-      <Stack flex spacing="XXXSmall" align="center" spaceAfter={spaceAfter}>
+      <Stack flex spacing="none" align="center" spaceAfter={spaceAfter}>
         {children}
-        <ButtonLink
-          iconLeft={<LinkIcon />}
-          title={`Link to heading: ${headingText}`}
-          href={`#${slugifiedText}`}
-          type="secondary"
-          compact
-        />
+        <StyledAnchor href={`#${slugifiedText}`} title={`Link to heading: ${headingText}`}>
+          <LinkIcon />
+        </StyledAnchor>
```

Comment:
It's better to do it right considering that you're refactoring this. I think `StyledAnchorWrapper` is the one that needs to be an anchor, not `StyledAnchor` (obviously you'd name them differently then), because elements that aren't focusable should not be hoverable. You can see that tabbing through these anchors is not good because they are invisible.

Reply:
nope, the anchor should be only for that hidden anchor, the tile should not be inside `a` tag. I doubt that they should be accessible via tab too. Check for instance [here](https://leerob.io/blog/style-guides-component-libraries-design-systems#what-is-a-style-guide). It should appear on heading hover, not the link, i will fix that.

Reply:
nope, only icon should be wrapped in anchor, if i got u right.

Reply:
I think the best would be to have in the way like Lee R has in his [blog](https://leerob.io/blog/monorepo-lerna-yarn-workspaces). I know that we don't use the rehype or remark plugin for autolink-headings, but our solution is quite similar